From 79aa2d43627c0c98c6ca8a81f6d8c636f5c8cdd7 Mon Sep 17 00:00:00 2001
From: Julien Bonjean <julien@bonjean.info>
Date: Sun, 24 Jan 2016 03:32:36 -0500
Subject: [PATCH] fix segmentation_id bug

This bugs causes the following error:
invalid literal for int() with base 10: 'None' Agent terminated!

Review: https://review.openstack.org/#/c/238485/
---
 .../plugins/ml2/drivers/openvswitch/agent/ovs_neutron_agent.py | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/usr/lib/python2.7/dist-packages/neutron/plugins/ml2/drivers/openvswitch/agent/ovs_neutron_agent.py b/usr/lib/python2.7/dist-packages/neutron/plugins/ml2/drivers/openvswitch/agent/ovs_neutron_agent.py
index b18716e..306cd48 100644
--- a/usr/lib/python2.7/dist-packages/neutron/plugins/ml2/drivers/openvswitch/agent/ovs_neutron_agent.py
+++ b/usr/lib/python2.7/dist-packages/neutron/plugins/ml2/drivers/openvswitch/agent/ovs_neutron_agent.py
@@ -353,11 +353,17 @@ class OVSNeutronAgent(sg_rpc.SecurityGroupAgentRpcCallbackMixin,
             net_uuid = local_vlan_map.get('net_uuid')
             if (net_uuid and net_uuid not in self.local_vlan_map
                 and local_vlan != DEAD_VLAN_TAG):
+                segmentation_id = local_vlan_map.get('segmentation_id')
+                if segmentation_id == 'None':
+                    # Backward compatible check when we used to store the
+                    # string 'None' in OVS
+                    segmentation_id = None
+                if segmentation_id is not None:
+                    segmentation_id = int(segmentation_id)
                 self.provision_local_vlan(local_vlan_map['net_uuid'],
                                           local_vlan_map['network_type'],
                                           local_vlan_map['physical_network'],
-                                          int(local_vlan_map[
-                                              'segmentation_id']),
+                                          segmentation_id,
                                           local_vlan)

     def setup_rpc(self):
--
2.7.0.rc3
